ilc/common/Slot/SpaSlot.js

```diff
@@ -0,0 +1,74 @@
+import {assert} from '../../client/utils';
+
+export class SpaSlot {
+    #applicationId;
+    #applicationName;
+    #slotName;
+    /**
+     * @property {IlcConfigRoot}
+     */
+    #configRoot
+
+    /**
+     * @constructor
+     * @param {{slotName: string, appName: string, appId: string}} slotName
+     * @param {IlcConfigRoot} rootConfig
+     */
+    constructor({applicationId, applicationName, slotName}, configRoot) {
+
+        assert(applicationId !== 'string', 'SpaSlot instance can not be initiated without applicationId');
+        assert(applicationName !== 'string', 'SpaSlot instance can not be initiated without applicationName');
+        assert(slotName !== 'string', 'SpaSlot instance can not be initiated without slotName');
```

Comment:
Those messages do not have all possible needed information, I would print in every message parameters that were specified, otherwise will be hard to debug

Reply:
I mean, Slot could be created for any app and route, right? By throwing error that something is defined without something does not give us understanding what app has error. So I would print initial object that passed in ctor to error to have additional information